File: chaeryeon823/BOJ_16953.js
// A->B

const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

rl.on("line", (line) => {
  let [a, b] = line.split(" ").map(Number);
  let answer = 1;

  while (b !== a) {
    let temp = b;
    if (b % 10 === 1) {
      b = Math.floor(b / 10);
      answer += 1;
    } else if (b % 2 === 0) {
      b = b / 2;
      answer += 1;
    }

    if (b === temp) {
      answer = -1;
      break;
    }
  }

  console.log(answer);
  rl.close();
});

File: julia8024/BOJ_16953.py
# 백준
# 실버2 A -> B

a, b = map(int, input().split())
r = 1

while (b != a):
    r += 1
    temp = b
    
    # b의 일의자리수가 1인 경우 -1
    if b%10 == 1:
        b //= 10
    # 짝수인 경우 2로 나누기
    elif b%2 == 0:
        b //= 2
    
    # 두 연산을 수행했음에도 값의 변화가 없는 경우
    # 무한루프이므로 탈출하고 -1 출력
    if temp == b:
        print(-1)
        break
else:
    print(r)
